internal/ingress/backend/endpoint.go

```diff
@@ -93,14 +146,72 @@ func (resolver *endpointResolver) resolveIP(ingress *extensions.Ingress, backend
 		return nil, fmt.Errorf("Unable to find service endpoints for %s: %v", serviceKey, err.Error())
 	}
 
+	podMap := map[string]*corev1.Pod{}
+	pods := resolver.store.GetServicePods(service.Spec.Selector)
+	for _, pod := range pods {
+		podMap[pod.Name] = pod
+	}
+	conditionTypePrefix := fmt.Sprintf("target-health.%s/", parser.AnnotationsPrefix)
+	conditionType := api.PodConditionType(fmt.Sprintf("%s%s_%s_%s", conditionTypePrefix, ingress.Name, service.Name, backend.ServicePort.String()))
+
 	var result []*elbv2.TargetDescription
 	for _, epSubset := range eps.Subsets {
 		for _, epPort := range epSubset.Ports {
 			// servicePort.Name is optional if there is only one port
 			if servicePort.Name != "" && servicePort.Name != epPort.Name {
 				continue
 			}
-			for _, epAddr := range epSubset.Addresses {
+			addresses := epSubset.Addresses
+			if service.Spec.PublishNotReadyAddresses {
```

Comment:
i don't think PublishNotReadyAddresses is used for this case.

Reply:
I see, according to the documentation it's mainly meant for DNS. We don't need this at the moement, I just thought it _might_ be useful for _someone_. I can remove it if you want.

Reply:
yeah, i think it's better to remove it since it's not our use case.